### ISSUE TITLE
Error 0x80070022 during cache of compressed package with multi-disk bundle

### ISSUE THREAD

**Issue by dan-olsen** [wixtoolset/issues#6050]
* Which version of WiX are you building with?

> 3.11.1.2318

* Which version of Visual Studio are you building with (if any)?

> 2017 v15.9.12

* Which version of the WiX Toolset Visual Studio Extension are you building with (if any)?

> 0.9.21.62588

* Which version of .NET are you building with?

> 4.0

* If the problem occurs when installing your packages built with WiX, what is the version of Windows the package is running on?

> Windows 7 and 10

* Describe the problem and the steps to reproduce it.

> We have a 2 disk install and we hook into the ResolveSource event to prompt for disk 2 for the final handful of packages located on disk 2. However, the last package we install is a exe package that has the Compressed attribute set to "true". Typically we don't have any problem with this and we get "Verified acquired payload" log messages for that package and it's payload (we never see "Acquiring package" messages though). In our latest release we have started seeing pop ups saying "Setup.exe - Wrong Volume: The wrong volume is in the drive. Please insert volume "disk 1 volume name" into drive D:" with Cancel, Try Again, Continue options. I am seeing a System event ID 26 in the event log "Application Popup". Also, this only seems to happen sometimes with no pattern emerging on when it happens. 

* Describe the behavior you expected and how it differed from the actual behavior.

> I would expect this to not happen or at least for there to be a log message regarding this.

Is there some attribute setting or event that I am handling?

If disk 1 is inserted and "Try Again" is pressed everything is ok. If disk 2 is left in and "Continue" is pressed the following error is encountered:

```
Error 0x80070022: Failed to read during cabinet extraction.
Error 0x80070022: Failed to extract all files from container, erf: 1:2:0
Error 0x80070022: Failed to wait for operation complete.
Error 0x80070022: Failed to open container.
Error 0x80070022: Failed to open container: WixAttachedContainer.
Failed to extract payloads from container: WixAttachedContainer to working path: %temp%\{GUID}\.cr\Setup.exe, error: 0x80070022.
Applied non-vital package: PackageId, encountered error: 0x80070022. Continuing...
```

Here is the snippet for the compressed exe package I have authored for reference:
```
<ExePackage Id="PackageId"
            DisplayName="Package Name"
            SourceFile="Some.exe"
            Compressed="yes"
            Vital="no"
            PerMachine="yes"
            DetectCondition="DetectCondition"
            InstallCommand="InstallCommand"
            RepairCommand="RepairCommand"
            UninstallCommand="UninstallCommand"
            SuppressSignatureVerification="yes" >
    <Payload SourceFile="some.dll"/>
    <Payload SourceFile="Some.exe.config"/>
</ExePackage>
```

**Comment by BMurri** [comment]
Please tell me that your bundle is not named `Setup.exe`

`WixAttachedContainer to working path: %temp%\{GUID}\.cr\Setup.exe, error: 0x80070022.`

**Comment by dan-olsen** [comment]
@BMurri It is not that was just for example.

**Comment by dan-olsen** [comment]
After looking at this a little more I figured out that in our last release we had 2 compressed ExePackages (this one and another that was earlier in the chain before we prompted for disk 2). I think it used to work because burn used to extract the WixAttachedContainer earlier in the cache process before we prompted for disk 2. And now with just the 1 compressed ExePackage it doesn't look for the WixAttachedContainer until after we have the user insert disk 2. However, this would only explain this pop up and error if it happened every install which it is not. It has happened a handful of times.

**Comment by BMurri** [comment]
It would be useful to see the entire logging for both successful (when this issue doesn't appear) and unsuccessful (when it does) as there may be some other difference that didn't appear important that turns out to be predictive.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by dan-olsen** [comment]
Actually it looks like it is always happening but it doesn't always have the pop up. It just silently logs the 0x80070022 error.

So it appears my thought on this being caused by the removal of the first compressed ExePackage is the cause.

**Comment by dan-olsen** [comment]
@BMurri I can't provide logs but it really does seem to be caused by what I said in my above comment. The issue I see is why do we get a 0x80070022 (ERROR_WRONG_DISK) error with/without the pop up instead of burn logging an error that the WixAttachedContainer can't be found.

**Comment by BMurri** [comment]
It literally logged (ERROR_WRONG_DISK) when trying to extract from the WixAttachedContainer. I agree that it would be helpful to have the logs include the error messages associated with the error codes, and there may even be one or more issues open to that effect, which would make human parsing of the log files potentially much more efficient/effective.

What isn't clear is why you get this state sometimes and not others.

**Comment by dan-olsen** [comment]
Agreed. The case where it just logs the error and moves on is fine. In our cause it's just silently logging because the package it is looking for is not vital.

I for sure know we saw the pop up when installing on Windows 7 32 bit. We are just not as certain if we ever saw the pop up when installing on Windows 10.

**Comment by barnson** [comment]
Burn should call `::SetErrorMode(SEM_FAILCRITICALERRORS)` which might help with this.
